lib/streamlit/caching/singleton_decorator.py

```diff
@@ -248,23 +256,26 @@ def __init__(self, key: str, display_name: str):
         self.display_name = display_name
         self._mem_cache: Dict[str, Any] = {}
         self._mem_cache_lock = threading.Lock()
+        self._message_cache: Dict[str, List[MsgData]] = {}
```

Comment:
I originally had both memo and singleton with separate dicts for values and messages, but changed memo to make persisting to disk easier. Potentially I should propagate that change to singleton as well, to reduce unnecessary differences between them